tensorflow/contrib/verbs/rdma.cc

```diff
@@ -471,8 +477,12 @@ void RdmaChannel::Connect(const RdmaAddress& remoteAddr) {
     attr.dest_qp_num = remoteAddr.qpn;
     attr.rq_psn = remoteAddr.psn;
     attr.max_dest_rd_atomic = 1;
-    attr.min_rnr_timer = 12;
-    attr.ah_attr.is_global = 0;
+    attr.min_rnr_timer = 12;   
+    attr.ah_attr.is_global = 1;
+    attr.ah_attr.grh.dgid.global.subnet_prefix = remoteAddr.snp;
+    attr.ah_attr.grh.dgid.global.interface_id = remoteAddr.iid;
+    attr.ah_attr.grh.flow_label = 0;
+    attr.ah_attr.grh.hop_limit = 3; 
     attr.ah_attr.dlid = remoteAddr.lid;
```

Comment:
@junshi15 I currently only have RoCE network. I can try to find a Infiniband network to test on.

@shamoya Since RoCEv2 packets have IP/UDP headers, they can be routed in multi-hop IP networks. The testbed I'm using have 2-tier Ethernet network (leaf-spine topology) with support for 802.1Qbb (data center bridging). I choose hop-limit=3 so that the RoCEv2 packets can be routed in a 2-tier network (leaf-switch --> spine-switch --> leaf-switch). I think it could be set higher to accommodate large-scale networks with more switching tiers.